### PR TITLE
apply comments

### PR DESCRIPTION
**Pull request by Shravanpaul**

### CODE REVIEW

File: utility/index.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            return Promise.reject(err);
+        }
+
+    }
+
+    async comparePass(password) {
+        try {
+            
+        } catch (err) {
+            console.log(err);
```
Comment (shekhartyagi26) [inline utility/index.js]: remove console.log

File: utility/index.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            return Promise.reject(err);
+        }
+
+    }
+
+    async comparePass(password) {
+        try {
+
```
Comment (shekhartyagi26) [inline utility/index.js]: remove extra spaces

File: utility/index.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    async bcryptPass(password) {
+        try {
+            let hashpass;
+            if (password) {
+                var salt = bcrypt.genSaltSync(2);
+                hashpass = await bcrypt.hashSync(password, salt);
+            }
+            return hashpass;
+        } catch (err) {
+            console.log(err);
```
Comment (shekhartyagi26) [inline utility/index.js]: remove console.log(),

File: utility/index.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+class Utility {
+    async bcryptPass(password) {
+        try {
+            let hashpass;
+            if (password) {
+                var salt = bcrypt.genSaltSync(2);
+                hashpass = await bcrypt.hashSync(password, salt);
+            }
+            return hashpass;
```
Comment (shekhartyagi26) [inline utility/index.js:14]: if password is not defined, then hashpass will return undefined, fix this

File: schema/userSchema.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  logIn
+};
+
+module.exports = UserValidator;
+
+function list () {
+  return {};
+}
+
+function me () {
```
Comment (shekhartyagi26) [inline schema/userSchema.js]: use the proper naming convention.

File: schema/userSchema.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        .string()
+        .alphanum()
+        .regex(/^(?=[a-f\d]{24}$)(\d+[a-f]|[a-f]+\d)/i, '_id')
+        .required()
+    }
+  };
+}
+
+function create () {
+  return {
```
Comment (shekhartyagi26) [inline schema/userSchema.js]: create a function where you just need to send the value and after that validate automatically.

File: schema/userSchema.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+const UserValidator = {
+  list,
+  me,
+  create,
+  logIn
+};
+
+module.exports = UserValidator;
+
+function list () {
```
Comment (shekhartyagi26) [inline schema/userSchema.js]: what's the use of this,

File: routes/userSchema.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+const UserController =require('../controller/user');
+const Validator = require('../schema/userSchema');
+exports.register = function (server, options, next) {
+    // instantiate controller
+    const userController = new UserController(server.db);
+
+    server.bind(userController);
+
+    // GET USER API
+    server.route({
```
Comment (shekhartyagi26) [inline routes/userSchema.js]: you call the controller from userSchema, do you think is that correct?

File: controller/book.js [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+// [PUT] /book
+function update (request, reply) {
+  const id = request.params.id;
+  const payload = request.payload;
+
+  this.model.findOneAndUpdate({_id: id}, {$set: payload}, {new: true})
+  .then((book) => {
+    reply(book);
```
Comment (shekhartyagi26) [inline controller/book.js]: it would be better if we use resolve instead of reply.

File: controller/book.js [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+  });
+}
+
+// [DELETE] /book
+function destroy (request, reply) {
+  const id = request.params.id;
+
+  this.model.remove({_id: id})
+  .then(() => {
+    reply();
```
Comment (shekhartyagi26) [inline controller/book.js]: what will be the success message here

File: controller/book.js (hunk 9 above)
Comment (Shravanpaul) [reply controller/book.js]: okk sir

File: controller/book.js (hunk 10 above)
Comment (Shravanpaul) [reply controller/book.js]: i just send status(202) no content

File: routes/userSchema.js (hunk 8 above)
Comment (Shravanpaul) [reply routes/userSchema.js]: I just changed it

File: schema/userSchema.js (hunk 7 above)
Comment (Shravanpaul) [reply schema/userSchema.js]: this is use for validate user inputs

File: utility/index.js (hunk 4 above)
Comment (Shravanpaul) [reply utility/index.js:14]: okk fine sir

File: utility/index.js (hunk 2 above)
Comment (Shravanpaul) [reply utility/index.js]: okk fine sir

File: utility/index.js (hunk 1 above)
Comment (Shravanpaul) [reply utility/index.js]: resolve

File: utility/index.js (hunk 2 above)
Comment (Shravanpaul) [reply utility/index.js]: resolve

File: utility/index.js (hunk 3 above)
Comment (Shravanpaul) [reply utility/index.js]: resolve

File: utility/index.js (hunk 4 above)
Comment (Shravanpaul) [reply utility/index.js:14]: resolve

File: schema/userSchema.js (hunk 5 above)
Comment (Shravanpaul) [reply schema/userSchema.js]: resolve

File: schema/userSchema.js (hunk 6 above)
Comment (Shravanpaul) [reply schema/userSchema.js]: this is check before passing to controller

**Pull request merged**